### PR TITLE
Bug fix/internal issue #605

### PR DESCRIPTION
**Pull request by gnusi**
- fix arangosearch recovery issue
- upgrade iresearch

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6696/

File: tests/js/server/recovery/view-arangosearch-link-populate-crash-after-release.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    testIResearchLinkPopulateNoRelease: function () {
+      var v = db._view('UnitTestsRecoveryView');
+      assertEqual(v.name(), 'UnitTestsRecoveryView');
+      assertEqual(v.type(), 'arangosearch');
+      var p = v.properties().links;
+      assertTrue(p.hasOwnProperty('UnitTestsRecoveryDummy'));
+      assertTrue(p.UnitTestsRecoveryDummy.includeAllFields);
+
+      var result = AQL_EXECUTE("FOR doc IN UnitTestsRecoveryView SEARCH doc.c >= 0 OPTIONS {waitForSync: true} COLLECT WITH COUNT INTO length RETURN length").json;
+      assertTrue(result[0] > 0);
```
Comment (jsteemann) [inline tests/js/server/recovery/view-arangosearch-link-populate-crash-after-release.js]: shouldn't this test always assert to find exactly 10,000 results?

File: tests/js/server/recovery/view-arangosearch-link-populate-no-sync.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    testIResearchLinkPopulateNoRelease: function () {
+      var v = db._view('UnitTestsRecoveryView');
+      assertEqual(v.name(), 'UnitTestsRecoveryView');
+      assertEqual(v.type(), 'arangosearch');
+      var p = v.properties().links;
+      assertTrue(p.hasOwnProperty('UnitTestsRecoveryDummy'));
+      assertTrue(p.UnitTestsRecoveryDummy.includeAllFields);
+
+      var result = AQL_EXECUTE("FOR doc IN UnitTestsRecoveryView SEARCH doc.c >= 0 OPTIONS {waitForSync: true} COLLECT WITH COUNT INTO length RETURN length").json;
+      assertTrue(result[0] > 0);
```
Comment (jsteemann) [inline tests/js/server/recovery/view-arangosearch-link-populate-no-sync.js]: same here: shouldn't this test assert to find exactly 10,000 results?

File: tests/IResearch/common.cpp [hunk 3]
```diff
@@ -177,22 +180,25 @@ struct CustomScorer : public irs::sort {
...
+      struct ScoreCtx : public irs::sort::score_ctx {
+        ScoreCtx(float_t score) : i(score) {}
 
         float_t i;
       };
 
-      return irs::sort::scorer::make<Scorer>(i);
+      return {
+        std::make_unique<ScoreCtx>(i),
+        [](const void* ctx, irs::byte_type* score_buf) NOEXCEPT {
```
Comment (jsteemann) [inline tests/IResearch/common.cpp]: I think we can use `noexcept` here (lower-case letters).

File: tests/IResearch/common.cpp [hunk 4]
```diff
@@ -102,22 +102,25 @@ struct BoostScorer : public irs::sort {
...
+      struct ScoreCtx : public irs::sort::score_ctx {
+        ScoreCtx(irs::boost_t score) : scr(score) {}
 
         irs::boost_t scr;
       };
 
-      return irs::sort::scorer::make<Scorer>(boost);
+      return {
+        std::make_unique<ScoreCtx>(boost),
+        [](const void* ctx, irs::byte_type* score_buf) NOEXCEPT {
```
Comment (jsteemann) [inline tests/IResearch/common.cpp]: I think we can use `noexcept` here (lower-case letters).

**Pull request merged**